### PR TITLE
[feat] Permissions / DB roles

### PR DESCRIPTION
**Pull request by joshuajeschek**

### CODE REVIEW

File: src/test/java/github/weichware10/util/db/PermissionsTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package github.weichware10.util.db;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import io.github.cdimascio.dotenv.Dotenv;
```
Comment (weichbot10) [inline src/test/java/github/weichware10/util/db/PermissionsTest.java:8]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'io.github.cdimascio.dotenv.Dotenv'

Comment (weichbot10) [inline src/test/java/github/weichware10/util/db/PermissionsTest.java:8]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'io.github.cdimascio.dotenv.Dotenv' import. Should be before 'org.junit.Test'.

File: src/test/java/github/weichware10/util/db/PermissionsTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+public class PermissionsTest {
```
Comment (weichbot10) [inline src/test/java/github/weichware10/util/db/PermissionsTest.java:10]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocTypeCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/util/db/DataBaseClient.java [hunk 3]
```diff
@@ -62,17 +64,21 @@ public DataBaseClient(String url, String user, String password, String schema) {
         this.configurations = new Configurations(this);
         this.trials = new Trials(this);
         this.datapoints = new Datapoints(this);
+        this.permissions = new Permissions(this);
     }
 
     /**
-     * Überprüft, ob Zugriff auf die spezifizierte Tabelle im Schema besteht.
+     * Überprüft, ob die spezifizierte Tabelle im schema existiert.
```
Comment (TheDarkDesync) [inline src/main/java/github/weichware10/util/db/DataBaseClient.java:71]:
```suggestion
     * Überprüft, ob die spezifizierte Tabelle im Schema existiert.
```

**Review by TheDarkDesync — APPROVED** [review]
image
works as intended

File: src/test/java/github/weichware10/util/db/PermissionsTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        assertTrue("test_subject should be subject", subjectClient.permissions.isSubject);
+        assertFalse("test_subject should not be spectator", subjectClient.permissions.isSpectator);
+
+        DataBaseClient spectatorClient = new DataBaseClient(
+                dotenv.get("DB_URL"),
+                "test_spectator",
+                dotenv.get("DB_PASSWORD"),
+                dotenv.get("DB_SCHEMA"));
+
+        assertFalse("test_spectator should be admin", spectatorClient.permissions.isAdmin);
```
Comment (grenoui11e) [inline src/test/java/github/weichware10/util/db/PermissionsTest.java:55]:
```suggestion
        assertFalse("test_spectator should not be admin", spectatorClient.permissions.isAdmin);
```

File: src/test/java/github/weichware10/util/db/PermissionsTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        assertFalse("test_spectator should be author", spectatorClient.permissions.isAuthor);
```
Comment (grenoui11e) [inline src/test/java/github/weichware10/util/db/PermissionsTest.java:56]:
```suggestion
        assertFalse("test_spectator should not be author", spectatorClient.permissions.isAuthor);
```

File: src/test/java/github/weichware10/util/db/PermissionsTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        assertFalse("test_spectator should be subject", spectatorClient.permissions.isSubject);
```
Comment (grenoui11e) [inline src/test/java/github/weichware10/util/db/PermissionsTest.java:57]:
```suggestion
        assertFalse("test_spectator should not be subject", spectatorClient.permissions.isSubject);
```

**Review by grenoui11e — APPROVED** [review]

**Pull request merged**